junit-platform-commons/src/main/java/org/junit/platform/commons/util/ClasspathScanner.java

```diff
@@ -170,10 +170,19 @@ private String determineSimpleClassName(Path classFile) {
 		return fileName.substring(0, fileName.length() - CLASS_FILE_SUFFIX.length());
 	}
 
-	private String determineSubpackageName(Path baseDir, Path classFile) {
+	private String determineSubpackageName(Path baseDir, String basePackageName, Path classFile) {
 		Path relativePath = baseDir.relativize(classFile.getParent());
 		String pathSeparator = baseDir.getFileSystem().getSeparator();
 		String subpackageName = relativePath.toString().replace(pathSeparator, PACKAGE_SEPARATOR_STRING);
+
+		if (StringUtils.isNotBlank(basePackageName) && subpackageName.startsWith(basePackageName)) {
+			if (subpackageName.length() == basePackageName.length()) {
+				subpackageName = "";
+			} else {
+				subpackageName = subpackageName.substring(basePackageName.length() + 1);
+			}
```

Comment:
I think this `if..else` could be shortened to a ternary statement, but it's not critical. :)